modules/diskq/qdisk.c

```diff
@@ -224,6 +224,47 @@ _truncate_file(QDisk *self, gint64 new_size)
   return success;
 }
 
+static gint64
+qdisk_get_lowest_used_queue_offset(QDisk *self)
+{
```

Comment:
Maybe the iteration does not look bad (you mentioned the `offsets[i]`), 
it's just it can be done nice.

Your suggestion isn't perfect, as in case of empty queues (offset = 0), it would result in `GINTMAX`.

Reply:
I will resolve this based on @Kokan's comment, although the return value will be 0 in case no memory queue have been used (positions are 0).

Reply:
Reworked based on @Kokan's comment.